File: 0231-power-of-two/0231-power-of-two.cs
public class Solution {
    
    bool solve(long ans, long n)
    {
        if(ans==n) {
            return true;
        }
        if(ans>n) {
            return false;
        }
        
        bool ok = solve(ans*2, n);
        
        return ok;
    }
    
    public bool IsPowerOfTwo(int n) {
        if(n<=0 || n == int.MaxValue) {
            return false;
        }
        return solve(1, n);
    }
}

File: 0231-power-of-two/0231-power-of-two.cpp
class Solution {
public:
    bool isPowerOfTwo(int n) {
        
        if(n<=0)return false;
        
        int cnt = 0;
        while(n!=0)
        {
            n = n&(n-1);
            cnt++;
        }
        
        return cnt==1;
    }
};